addons/mail/controllers/thread.py

```diff
@@ -147,6 +147,7 @@ def _prepare_post_data(self, post_data, thread, partner_emails=None, **kwargs):
         if not request.env.user._is_internal():
             partners = partners & self._filter_message_post_partners(thread, partners)
         post_data["partner_ids"] = partners.ids
+        post_data["thread_model"] = thread._name
```

Comment:
There is no need to add this param here. As you can always access the model of the thread when sending emails itself.